### DISCUSSION TITLE
Importing tailwind forms as an esm module

### DISCUSSION THREAD

**Discussion by JosephO1992** [tailwindlabs/tailwindcss#8946] [Help] [answered]
I'm using ESM modules in my current project.

I'm trying to implement the forms plugin, but I get an error as I can't use require('@tailwindcss/forms') as require isn't defined in the ES module scope. 

Does anyone have any ideas how to get this running?

Error below:

ReferenceError: require is not defined in ES module scope, you can use import instead
This file is being treated as an ES module because it has a '.js' file extension and '/Users/joeoreilly/Documents/coding/notanothercrm/package.json' contains "type": "module". To treat it as a CommonJS script, rename it to use the '.cjs' file extension.

Tailwind.config.js :

export default {
	content: ["./index.html", "./src/**/*.{vue,js,ts,jsx,tsx}"],
	theme: {
		extend: {},
	},
	plugins: [require("@tailwindcss/forms")],
};

Thanks

**Comment by wongjn** [comment, marked as answer]
ESM module syntax and CommonJS syntax is being mixed here. Tailwind requires the configuration file to be in CommonJS format:

```diff
-export default {
+module.exports = {
   content: ["./index.html", "./src/**/*.{vue,js,ts,jsx,tsx}"],
   theme: {
     extend: {},
   },
   plugins: [require("@tailwindcss/forms")],
 };
```

**Reply by damianobarbati** [reply]
@wrongjn this is not working anyway:
```sh
Server running at http://localhost:1234
🚨 Build failed.

@parcel/transformer-postcss: require() of ES Module
/Users/damians/Desktop/box696/tailwind.config.js from
/Users/damians/Desktop/box696/node_modules/tailwindcss/lib/lib/setupTrackingContext.js
 not supported.
Instead change the require of tailwind.config.js in
/Users/damians/Desktop/box696/node_modules/tailwindcss/lib/lib/setupTrackingContext.js
 to a dynamic import() which is available in all CommonJS modules.
```

**Reply by wongjn** [reply]
Indeed – Tailwind requires the configuration file to be in CommonJS format. I have updated the answer.
